src/libdredd/src/mutation_replace_expr.cc

```diff
@@ -77,12 +88,25 @@ std::string MutationReplaceExpr::GetFunctionName(
 bool MutationReplaceExpr::ExprIsEquivalentTo(const clang::Expr& expr,
                                              int constant,
                                              clang::ASTContext& ast_context) {
-  clang::Expr::EvalResult int_result;
+  clang::Expr::EvalResult int_eval_result;
   if (expr.getType()->isIntegerType() &&
-      expr.EvaluateAsInt(int_result, ast_context)) {
-    return llvm::APSInt::isSameValue(int_result.Val.getInt(),
+      expr.EvaluateAsInt(int_eval_result, ast_context)) {
+    return llvm::APSInt::isSameValue(int_eval_result.Val.getInt(),
                                      llvm::APSInt::get(constant));
   }
+
+  llvm::APFloat float_eval_result(static_cast<double>(0));
+  if (expr.getType()->isFloatingType() &&
+      expr.EvaluateAsFloat(float_eval_result, ast_context)) {
+    return float_eval_result.isExactlyValue(constant);
+  }
+
+  bool result = false;
```

Comment:
Rename `result` to `bool_eval_result` for consistency (even though it has a different type), and to avoid breaking the convention that `result` is used for the thing that will be returned.

Reply:
Done.